services/submission/src/main/java/app/coronawarn/server/services/submission/ServerApplication.java

```diff
@@ -59,6 +64,26 @@ TimedAspect timedAspect(MeterRegistry registry) {
     return new TimedAspect(registry);
   }
 
+  @Bean
+  public WaitOnTomcatToProcessPendingRequests gracefulShutdown() {
+    return new WaitOnTomcatToProcessPendingRequests();
+  }
+
+  /**
+   * Customize TomcatServletWebServerFactory by adding {@link WaitOnTomcatToProcessPendingRequests}.
+   */
+  @Bean
+  public ConfigurableServletWebServerFactory webServerFactory(WaitOnTomcatToProcessPendingRequests gracefulShutdown) {
+    TomcatServletWebServerFactory factory = new TomcatServletWebServerFactory();
+    factory.addConnectorCustomizers(gracefulShutdown);
+    return factory;
+  }
+
+  @PreDestroy
+  public void onDestroy() {
+    LogManager.shutdown();
```

Comment:
If we shut down the LogManager on "pre-destroy", does this mean that we can not log stuff anymore *during* destroy?

Reply:
I believe we should have a consistent implementation for this - the distribution service has the same functionality, but not using `javax.*` but instead spring boot style with `DisposableBean`. Let's do the same here.

https://github.com/corona-warn-app/cwa-server/blob/f41fcbad05acf1e0581dc7a4c8bac5cf76567620/services/distribution/src/main/java/app/coronawarn/server/services/distribution/Application.java#L58-L65

Reply:
Not an answer to the question, but I assume this relates to the same underlying issue as in [#589](https://github.com/corona-warn-app/cwa-server/issues/589).

Reply:
available [shutdown-callbacks](https://www.baeldung.com/spring-shutdown-callbacks)

Destroy methods are called in the order:
1. Methods annotated with @PreDestroy
1. destroy() as defined by the DisposableBean callback interface
1. A custom configured destroy() method

So which approach, has to be followed, when there are multiple ways of doing things?

Side note @pithumke: distribution service 
[logger.error("Application terminated abnormally.");](https://github.com/corona-warn-app/cwa-server/blob/master/services/distribution/src/main/java/app/coronawarn/server/services/distribution/Application.java#L72)
I guess this won't work, because `SpringApplication.exit(appContext);` earlier triggers the destroy method, which shuts down the logmanager...

Reply:
- `@PreDestroy` is somewhat deprecated with Java 11 (not sure which dependency makes it available in the submission module right now).
- Distribution module already uses Spring's `DisposableBean` interface as a non-deprecated way of achieving the desired functionality.

Reply:
now using `DisposableBean`